action.yml

```diff
@@ -58,9 +61,52 @@ runs:
               }
             ]
           }")
-          
-        host=$(echo $branch | jq --raw-output '.endpoints[0].host')
-        branch_id=$(echo $branch | jq --raw-output '.endpoints[0].branch_id')
+
+        if [[ $branch == *"already exists"* ]]; then
+
+        # Get the branch id by its name. We list all branches and filter by name
+        branch_id=$(curl --silent \
+            "https://console.neon.tech/api/v2/projects/${PROJECT_ID}/branches" \
+            --header "Accept: application/json" \
+            --header "Content-Type: application/json" \
+            --header "Authorization: Bearer ${API_KEY}" \
+            | jq -r .branches \
+            | jq -c '.[] | select(.name | contains("'${BRANCH_NAME}'")) .id' \
+            | jq -r \
+            ) \
+
+
+        # Get the list of endpoints for the branch by its id. 
+        endpoints=$(curl --silent \
+            "https://console.neon.tech/api/v2/projects/${PROJECT_ID}/branches/${branch_id}/endpoints" \
+            --header "Accept: application/json" \
+            --header "Content-Type: application/json" \
+            --header "Authorization: Bearer ${API_KEY}" \
+            ) \
+
+        endpoint_id=$(echo $endpoints | jq --raw-output '.endpoints[0].host' | cut -d'.' -f1)
+        region=$(echo $endpoints | jq --raw-output '.endpoints[0].host' | cut -d'.' -f2)
+        cloud_provider=$(echo $endpoints | jq --raw-output '.endpoints[0].host' | cut -d'.' -f3)
+        host=$(echo $endpoints | jq --raw-output '.endpoints[0].host')
+        branch_id=$(echo $endpoints | jq --raw-output '.endpoints[0].branch_id')
+
         echo "host=${host}" >> $GITHUB_OUTPUT
+        echo "endpoint_id=${endpoint_id}" >> $GITHUB_OUTPUT
         echo "branch_id=${branch_id}" >> $GITHUB_OUTPUT
-        echo "db_url=postgres://${PGUSERNAME}:${PGPASSWORD}@${host}" >> $GITHUB_OUTPUT
+        echo "db_url=postgres://${PGUSERNAME}:${PGPASSWORD}@${endpoint_id}.${region}.${cloud_provider}.neon.tech" >> $GITHUB_OUTPUT
```

Comment:
why not just use host?

Reply:
Just to be consistent with the other variable